### DISCUSSION TITLE
Verenmq behavior on retries without disconnection

### DISCUSSION THREAD

**Discussion by shivamkm07** [vernemq/vernemq#2058] [Q&A]
Vernemq states that it supports full MQTT 5.0 specification as per https://vernemq.com/mqttv5.html. MQTT5 clearly states that "Clients and Servers MUST NOT resend messages at any other time other than reconnection" (https://docs.oasis-open.org/mqtt/mqtt/v5.0/os/mqtt-v5.0-os.html#_Toc3901238)
However, vernemq currently seems to be retrying on unacked messages with QOS 1, after a fixed timeout (Tested with v1.11.0). It is also mentioned here: https://vernemq.com/intro/mqtt-primer/quality_of_service.html#at-least-once-qos-1. This is fine with MQTT 3 but not with MQTT 5.  So is it being tracked to be removed from vernemq in future releases?

Please note that we (https://github.com/dapr/) are using MQTT as Pubsub component (https://docs.dapr.io/reference/components-reference/supported-pubsub/setup-mqtt/) with Vernemq as broker. Dapr provides at least once guarantee for messages published with qos > 1. So if vernemq decides to deprecate retrying for unacked messages as per MQTT5, it would violate our guarantee and in the worst case we might need to deprecate our MQTT component.

**Comment by ioolkos** [comment]
@shivamkm07 Thanks. Are you asking whether we plan to deprecate retrying for **all** MQTT protocol version? Certainly not, as version 3 and 4 (3.1.1) require the feature. 
We'll have to think about what to do for v5, it seems. I have not seen any complaints about retries in v5 so far.

Thanks for using Verne! (it'd be great if you'd mention and link Verne in your docs, as you do with other brokers ;)

**Reply by shivamkm07** [reply]
Great, so even if retries are deprecated, they would be only done in v5 and not in v3 and v4. That would be good for us. Thanks!

Will definitely add and link Verne in our docs. Thanks for pointing out.
